backend/layers/persistence/persistence.py

```diff
@@ -0,0 +1,153 @@
+        pass
+
+    def get_all_versions_for_collection(self, collection_id: str) -> List[CollectionVersion]:
+        """
+        Retrieves all versions for a specific collections, without filtering
+        """
+        pass
+
+    def get_all_collections(self) -> Iterable[CollectionVersion]:
+        """
+        Retrieves all collections.
+        # TODO: add a filtering object.
+        """
+        pass
+
+    def delete_canonical_collection(self, collection_id: str) -> None:
+        """
+        Deletes (tombstones) a canonical collection.
+        """
+        pass
+
+    def save_collection_metadata(self, version_id: str, collection_metadata: CollectionMetadata) -> None:
+        """
+        Saves collection metadata for a collection version
+        """
+        pass
+
+    def save_collection_publisher_metadata(self, version_id: str, publisher_metadata: dict) -> None:
+        """
+        Saves publisher metadata for a collection version
+        """
+        pass
+
+    def add_collection_version(self, collection_id: str) -> CollectionVersion:
+        """
+        Adds a collection version to an existing canonical collection.
+        Returns the new version.
+        """
+        pass
+
+    def delete_collection_version(self, version_id: str) -> None:
+        """
+        Deletes a collection version.
+        """
+        pass
+
+    def finalize_collection_version(self, collection_id: str, version_id: str) -> None:
+        """
+        Finalizes a collection version. This is equivalent to calling:
+        1. update_collection_version_mapping
+        2. set_collection_version_published_at
+        """
+        pass
+
+    def update_collection_version_mapping(self, collection_id: str, version_id: str) -> None:
+        """
+        Updates the mapping between the canonical collection `collection_id` and its `version_id`
+        """
+        pass
+
+    def set_collection_version_published_at(self, version_id: str, published_at: datetime) -> None:
+        """
+        Sets the `published_at` flag for a collection version
+        """
+        pass
+
+    def get_dataset_version(self, dataset_version_id: str) -> DatasetVersion:
+        """
+        Returns a dataset version by id.
+        """
+        pass
+
+    def get_all_versions_for_dataset(self, dataset_id: str) -> List[DatasetVersion]:
+        """
+        Returns all dataset versions for a canonical dataset_id
+        """
+        pass
+
+    def get_all_datasets(self) -> Iterable[DatasetVersion]:  # TODO: add filters if needed
+        """
+        Returns all dataset versions.
+        # TODO: Add filtering
+        """
+        pass
+
+    def get_dataset_artifacts(self, dataset_version_id: str) -> List[DatasetArtifact]:
+        """
+        Returns all the artifacts for a specific dataset version
+        """
+        pass
+
+    def initialize_canonical_dataset(self, collection_version_id: str, dataset_metadata: DatasetMetadata) -> DatasetVersion:
```

Comment:
```suggestion
    def create_canonical_dataset(self, collection_version_id: str, dataset_metadata: DatasetMetadata) -> DatasetVersion:
```
to (now) be consistent with collection